front-end/src/actions/teamMembers.js

```diff
@@ -1,126 +1,68 @@
-import {
-  FETCH_TEAM_MEMBERS_REQUEST, FETCH_TEAM_MEMBERS_SUCCESS, FETCH_TEAM_MEMBERS_FAILURE,
-  CREATE_TEAM_MEMBER_REQUEST, CREATE_TEAM_MEMBER_SUCCESS, CREATE_TEAM_MEMBER_FAILURE,
-  DELETE_TEAM_MEMBER_REQUEST, DELETE_TEAM_MEMBER_SUCCESS, DELETE_TEAM_MEMBER_FAILURE,
-  UPDATE_TEAM_MEMBER_REQUEST, UPDATE_TEAM_MEMBER_SUCCESS, UPDATE_TEAM_MEMBER_FAILURE,
-  CHANGE_TEAM_REQUEST, CHANGE_TEAM_SUCCESS, CHANGE_TEAM_FAILURE
-} from '../actionTypes'
-
-import { thunkCreator } from './utils'
-
-export const fetchTeamMembers = () => thunkCreator({
-  types: [
-    FETCH_TEAM_MEMBERS_REQUEST,
-    FETCH_TEAM_MEMBERS_SUCCESS,
-    FETCH_TEAM_MEMBERS_FAILURE
-  ],
-
-  promise: fetch('/api/teamMembers')
-             .then(response => response.json())
-});
-
-const _createTeamMember = (teamMember) => thunkCreator({
-  types: [
-    CREATE_TEAM_MEMBER_REQUEST,
-    CREATE_TEAM_MEMBER_SUCCESS,
-    CREATE_TEAM_MEMBER_FAILURE
-  ],
-
-  promise: fetch('/api/teamMembers', {
-    method: 'POST',
-    headers: {
-      'Accept': 'application/json',
-      'Content-Type': 'application/json'
-    },
-    body: JSON.stringify({
-      ...teamMember
-    })
-  })
-  .then(response => response.json())
+import * as types from '../actionTypes';
+import { fetchData, fetchDataSuccess, fetchDataError } from './fetch';
+import API from '../utils/api';
+import uuid from 'uuid';
+
+export const fetchTeamMembers = teamMembers => ({
+	type: types.FETCH_TEAM_MEMBERS,
+	teamMembers
 });
 
-const _updateTeamMember = (teamMember) => thunkCreator({
-  types: [
-    UPDATE_TEAM_MEMBER_REQUEST,
-    UPDATE_TEAM_MEMBER_SUCCESS,
-    UPDATE_TEAM_MEMBER_FAILURE
-  ],
-
-  promise: fetch('/api/teamMembers', {
-    method: 'PUT',
-    headers: {
-      'Accept': 'application/json',
-      'Content-Type': 'application/json'
-    },
-    body: JSON.stringify({
-      ...teamMember
-    })
-  })
-  .then(response => response.json())
+export const addTeamMember = teamMember => dispatch => ({
+	type: types.CREATE_TEAM_MEMBER,
+	teamMember: {
+		...teamMember,
+		id: uuid()
+	}
 });
 
-export const createTeamMember = (teamMember) => (dispatch) =>
-  _createTeamMember(teamMember)(dispatch)
+export const handleFetchTeamMember = id => dispatch => {
+	dispatch(fetchData())
+	API.getTeamMemeber(id)
+	.then(res => {
+		dispatch(fetchTeamMember(res.data))
+		dispatch(fetchDataSuccess());
+	});
+}
+
+export const fetchTeamMember = teamMember => ({ 
+	type: types.FETCH_TEAM_MEMBER, 
+	teamMember
+})
+
+
+export const createTeamMember = (teamMember) => (dispatch) => {
```

Comment:
Rename to handleCreateTeamMember for consistency with handleFetchTeamMember.  It'll show the difference between the redux action to be dispatched and the function to be called which actually makes the request to the API.